src/controllers/reservationController.js

```diff
@@ -0,0 +1,76 @@
+import mongoose from 'mongoose';
+import { Reservation } from '../models/Reservation.js';
+import { RealEstateErrors } from '../utils/ErrorHandler.js';
+
+export const createReservation = async (req, res, next) => {
+  try {
+    const reservation = new Reservation(req.body);
+    await reservation.save();
+
+    const { _id, listingId, startDate, endDate } = reservation;
+    res.status(200).send({ id: _id, listingId, startDate, endDate });
+  } catch {
+    next(new RealEstateErrors());
+  }
+};
+
+export const getAllReservations = async (req, res, next) => {
+  try {
+    const reservations = await Reservation.find();
```

Comment:
Pass an empty object as it is better practise.
```suggestion
    const reservations = await Reservation.find({});
```